evm-adapters/src/sputnik/cheatcodes/debugger.rs

```diff
@@ -0,0 +1,383 @@
+    pub push_bytes: Option<Vec<u8>>,
+    pub ic: usize,
+}
+
+impl Default for DebugStep {
+    fn default() -> Self {
+        Self {
+            pc: 0,
+            stack: vec![],
+            memory: Memory::new(0),
+            op: OpCode(Opcode::INVALID, None),
+            push_bytes: None,
+            ic: 0,
+        }
+    }
+}
+
+impl DebugStep {
+    pub fn pretty_opcode(&self) -> String {
+        if let Some(push_bytes) = &self.push_bytes {
+            format!("{}(0x{})", self.op, hex::encode(push_bytes))
+        } else {
+            self.op.to_string()
+        }
+    }
+}
+
+impl Display for DebugStep {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(push_bytes) = &self.push_bytes {
+            write!(
+                f,
+                "pc: {:?}\nop: {}(0x{})\nstack: {:#?}\nmemory: 0x{}\n\n",
+                self.pc,
+                self.op,
+                hex::encode(push_bytes),
+                self.stack,
+                hex::encode(self.memory.data())
+            )
+        } else {
+            write!(
+                f,
+                "pc: {:?}\nop: {}\nstack: {:#?}\nmemory: 0x{}\n\n",
+                self.pc,
+                self.op,
+                self.stack,
+                hex::encode(self.memory.data())
+            )
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum CheatOp {
+    ROLL,
+    WARP,
+    FEE,
+    STORE,
+    LOAD,
+    FFI,
+    ADDR,
+    SIGN,
+    PRANK,
+    STARTPRANK,
+    STOPPRANK,
+    DEAL,
+    ETCH,
+    EXPECTREVERT,
+    RECORD,
+    ACCESSES,
+    EXPECTEMIT,
+}
+
+impl From<CheatOp> for OpCode {
+    fn from(cheat: CheatOp) -> OpCode {
+        OpCode(Opcode(0x0C), Some(cheat))
+    }
+}
+
+impl CheatOp {
+    pub const fn name(&self) -> &'static str {
+        match self {
+            CheatOp::ROLL => "VM_ROLL",
+            CheatOp::WARP => "VM_WARP",
+            CheatOp::FEE => "VM_FEE",
+            CheatOp::STORE => "VM_STORE",
+            CheatOp::LOAD => "VM_LOAD",
+            CheatOp::FFI => "VM_FFI",
+            CheatOp::ADDR => "VM_ADDR",
+            CheatOp::SIGN => "VM_SIGN",
+            CheatOp::PRANK => "VM_PRANK",
+            CheatOp::STARTPRANK => "VM_STARTPRANK",
+            CheatOp::STOPPRANK => "VM_STOPPRANK",
+            CheatOp::DEAL => "VM_DEAL",
+            CheatOp::ETCH => "VM_ETCH",
+            CheatOp::EXPECTREVERT => "VM_EXPECTREVERT",
+            CheatOp::RECORD => "VM_RECORD",
+            CheatOp::ACCESSES => "VM_ACCESSES",
+            CheatOp::EXPECTEMIT => "VM_EXPECTEMIT",
+        }
+    }
+}
+
+impl Default for CheatOp {
+    fn default() -> Self {
+        CheatOp::ROLL
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct OpCode(pub Opcode, pub Option<CheatOp>);
+
+impl From<Opcode> for OpCode {
+    fn from(op: Opcode) -> OpCode {
+        OpCode(op, None)
+    }
+}
+
+impl OpCode {
+    pub const fn name(&self) -> &'static str {
+        match self.0 {
```

Comment:
we have some non-matching names because sputnik's opcodes aren't up to date with current naming conventions, i.e.: `Opcode::SUICIDE` needs to map to `"SELFDESTRUCT"`